### PR TITLE
Add automatic timestamps to appended notes for Update Command

### LINKED ISSUE #96: Add Auto-Timestamps for Appended Notes for 'Update' Command

### PR DESCRIPTION
**Pull request by NigelGM**
Closes #96 

This PR implements a note-appending feature that automatically timestamps updates via the update command, enhancing clinical traceability.

Key Features

- New Prefix: Added an/ for appending to existing notes as done in #98 

- Automatic Timestamps: Prefixes are appended with [dd MMM HH:mm] (e.g., [24 Mar 05:30]).

- Smart Logic: Handles empty notes and placeholders (-) without extra newlines.

Validation: Ensures the combined note length stays within constraints.

Quality Assurance

- UI Formatting: Updated PersonListCard.fxml to ensure notes wrap properly.

Note for Reviewers
This branch is based on feature-append-notes. Please merge PR #98 first to isolate the timestamp-specific changes here.

### CODE REVIEW

**Review by e1384467 — COMMENTED** [review]
I left some questions and suggestions

File: src/main/java/seedu/address/model/person/Notes.java [hunk 1]
```diff
@@ -29,6 +32,34 @@ public Notes(String notes) {
...
+     * @return A new Notes object containing the combined text.
+     */
+    public Notes append(Notes additionalNotes) {
+        // 1. NO-OP GUARD: Safely ignore empty strings
+        if (additionalNotes.value.trim().isEmpty()) {
+            return this;
+        }
+
+        // 2. Generate the timestamp internally
+        String timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("dd MMM HH:mm"));
```
Comment (niikseah) [inline src/main/java/seedu/address/model/person/Notes.java]: I like the timestamp feature! Notes.append() calls LocalDateTime.now() internally, and the tests also compute the timestamp separately. Would it be possible to inject a singular source or pass the computed timestamp in so the behaviour is deterministic and we avoid edge-case failures around minute boundaries?
